aiida/orm/utils/loaders.py

```diff
@@ -427,7 +465,7 @@ def _get_query_builder_label_identifier(cls, identifier, classes):
             raise ValueError('the identifier needs to be a string')
 
         builder = QueryBuilder()
-        builder.append(cls=classes, tag='code', project=['*'], filters={'label': {'==': label}})
+        builder.append(cls=classes, tag='code', project=project, filters={'label': {'==': label}})
 
         if machinename:
             builder.append(Computer, filters={'name': {'==': machinename}}, with_node='code')
```

Comment:
As a note for the future, you might always want to append the Computer, just put the filter if you have a `machine name`, project the computer name, and then when you return (in general, for all _get_query_... functions) return also a string to say how to construct the output string from the projected values, e.g. doing a `.format(*projected_values)` (e.g. in general this will be `"{0}"` and here it would be `"{0}@{1}"`).